changelog and more info pages update (#243)
OctoMind-dev/mintlify-docs: changelog.mdx

icon: "sparkles"
---

## 2024-08-09

- `UI tweak` for first time users: Test result details of user's first test run are visible directly in project overview.

<Frame caption="First time users project overview with test result detail, 08/2024">
  <img
    src="images/changelog/first-report.png"
    alt="First time users project overview with test result detail"
  />
</Frame>

## 2024-08-08

- `Save & run`: New button to **run and save** an edited test case at once. You can click on the `run only` button if you want to validate the test first before saving it.

<Frame caption="'save & run' the test to validate and save changes, 08/2024">
  <img
    src="images/editing/save-run.png"
    alt="Save and run the test case"
    width="300px"
  />
</Frame>

<div class="mt-8" />

<Frame caption="'run only' to validate the test before saving it, 08/2024">
  <img
    src="images/editing/save-run-only.png"
    alt="Run only to validate the test before saving it"
    width="300px"
  />
</Frame>

## 2024-08-06

> - `AI suggesting more tests`: Similar to the AI discovery at set-up, we expanded the AI discovery feature to “spawn” individual test cases.

Reviewer: maybe just: to spawn **more** test cases?

<Frame caption="Have the AI agent suggest more test cases, 08/2024">
  <img
    src="images/setup/setup-11-suggest-more.png"
    alt="have AI suggest more test cases"
  />
</Frame>

> - `Open status for new test cases`: In order to mark test cases coming out of **suggest more** as new (until you've opened them for the first time), we introduced a new column **open status** to properly persist the state.

Reviewer: "... to properly persist state" !?!? why not just we introduce a new column.

Author: took it word by word from the release, sounded like we know what we were doing. I'll change it.

> - `Open status for new test cases`: In order to mark test cases coming out of **suggest more** as new (until you've opened them for the first time), we introduced a new column **open status** to properly persist the state.

Author: ```suggestion
- `Open status for new test cases`: In order to mark test cases coming out of **suggest more** as new (until you've opened them for the first time), we introduced a new column **open status** .
```

<Frame caption="Open status on new suggested tests, 08/2024">
  <img

>     src="images/changelog/open-status-proposals.png"

Reviewer: in the open status proposal the test case list view is inconsistent to the one above: two different suggest more buttons. consider to change the  open-status-proposals.png to show current state.

Author: yeah, those were two different states. I covered it up in figma. will do in the next commit.

- `Test case filters`: You can filter your tests based on the test statuses, now.

  <Frame caption="test case filters, 07/2024">
    <img
      src="/images/changelog/filters.png"
      alt="test case filters"
      width={400}
    />
  </Frame>

## 2024-07-11

- `Automatic login and cookie banner tests` at sign-up. When you sign up to Octomind or create new project (add a new URL):

1. AI agent goes through your site and checks for cookie banners.
2. If a cookie banner is present, it creates a cookie banner test.
3. AI agent checks if you have a login at your site.
4. If you do, it asks you for test credentials.
5. You give it your test credentials.
6. AI agent auto-generates a login test.
7. The cookie and the login tests will be populated into all new auto-generated tests as dependencies.

<Frame caption="AI agent asks for test credentials to auto-generate and run a login test, screenshot 07/2024">
  <img
    src="/images/setup/setup-5-login-credentials.png"
    alt="AI agent asks for login test credentials, screenshot 07/2024"
  />
</Frame>

## 2024-07-04

- `Mobile + tablet use`: Test case editing works better on smaller screens now.

## 2024-07-01

> - `Agent benchmark automation`: Every major change in the codebase impacts the AI agent performance. We get an automated ping whether the agent does what it icons supposed to, now.

Author: ```suggestion
- `Agent benchmark automation`: Every major change in the codebase impacts the AI agent performance. We get an automated ping whether the agent does what it's supposed to do, now.
```

> - `Agent benchmark automation`: Every major change in the codebase impacts the AI agent performance. We get an automated ping whether the agent does what it icons supposed to, now.

Reviewer: whether the agent does what it _icons_ supposed to, now.   ?!?

Author: whaha... oops
